netket/optimizer/qgt/qgt_jacobian_pytree_logic.py

```diff
@@ -114,7 +114,8 @@ def _divide_by_sqrt_n_samp(oks, samples):
     divide Oⱼₖ by √n
     """
     n_samp = samples.shape[0] * mpi.n_nodes  # MPI
-    return jax.tree_map(lambda x: x / np.sqrt(n_samp, dtype=x.dtype), oks)
+    sqrt_n = float(np.sqrt(n_samp, dtype=x.dtype))  # enforce weak type
```

Comment:
```suggestion
    sqrt_n = math.sqrt(n_samp)  # enforce weak type
```
(shorter but requires importing math, usually using this in my own code)

Reply:
thanks! that's a good idea